oar/core/jenkins_helper.py

```diff
@@ -137,3 +99,78 @@ def get_job_status(self, url, job_name, build_number):
             raise JenkinsHelperException(
                 f"get job{job_name}:{build_number} status failed"
             ) from e
+
+    def pre_check_build_queue(self,job_name):
+        """
+        check the job if it's in queue
+        """
+        try:
+            server = jenkins.Jenkins(
+                self._cs.get_jenkins_server(),
+                username=self._cs.get_jenkins_username(),
+                password=self._cs.get_jenkins_token(),
+            )
+            queue_info = server.get_queue_info()
+            for i in range(len(queue_info)):
```

Comment:
@wewang58 I think the loop logic can be refined. we don't have to use array index to exclude few items. another point, we should filter out the corresponding item with property `task` first. `blocked` is optional. if one job is in queue, we don't want the user to submit another one in parallel